### ISSUE TITLE
Has this library been deprecated?

### ISSUE THREAD

**Issue by grantkemp** [google/google-api-objectivec-client#210]
I just saw in the examples' readme  that this has been deprecated in favour of client for rest?

Is this the case?

**Comment by thomasvl** [comment]
It is being phased out.  There is a _Note_ in the readme that shows up on [main page](https://github.com/google/google-api-objectivec-client), and on all the other wiki pages pointing at the newer library.

**Comment by yoasha** [comment]
It's too bad that the life time of each API your release is relatively so short... I bet that many developers would appreciate a better backward compatibility of your APIs.

Any suggestion of when google-api-objectivec-client will be totally deprecated and inaccessible? Need to plan ahead.

**Comment by thomasvl** [comment]
Not sure I follow the first comment?  What is relatively short?

The issue is the server teams want to phase out JSON-RPC support (and just have REST with JSON).  Later this year we will likely flip the messaging to be stronger.  The REST version is pretty close api wise, but there were some differences in the server interfaces for some services.  We have a [migration doc](https://github.com/google/google-api-objectivec-client-for-rest/wiki/Migrating-From-GTL-to-GTLR) to help with the transition.

**Comment by yoasha** [comment]
Thomas, thanks for the clarification and for the link to the migration doc!

I guess you have good reasons to switch to a new library. I was just wondering why the protocol changes weren't integrated into the existing library.

**Comment by thomasvl** [comment]
The main problem the service apis themselves-how servers exposed things to JSON-RPC wasn't a 100% match to how they are exposed to REST+JSON.  So the core library could have been pretty close to api compatible, but the generated parts that talks to services wasn't under our control, and since it is generated, it isn't really something we can try to provide mappings.

We spent a long time trying to figure out what to do there, and did consider just making it a major version release that broke things.  What pushed us finally to the new project (and prefix) is it lets developers migrate on their own schedules even if they have dependencies.  Say you used the library and something you depend on also uses it (for another service/api).  If we simply moved this library at a major version, you would have to update your code at the same time you dependency does. But buy making it a new project, those two updates can be decoupled.  You and your dependency can move independently.
